### DISCUSSION TITLE
`glvnd` container with cuda

### DISCUSSION THREAD

**Discussion by Besler** [Kitware/trame#217] [Q&A] [answered]
I'm running a torch model through trame in a docker container. Could a `glvnd-cuda` tag be provided through the `nvidia/cuda:11.2.0-cudnn8-runtime-ubuntu20.04` base image? Alternatively, I could install the cuda/cudnn dependencies over the `glvnd` layer or manage the trame image myself?

**Comment by jourdain** [comment]
If you can `install the cuda/cudnn dependencies over the glvnd` that would be great. 

If not, we might be able to add yet another image but if we can prevent that, it would be great... I don't want to blow the matrix of builds... 

Out of curiosity,  are you using the rendering capability of the image with VTK or so, or not at all?

**Comment by Besler** [comment, marked as answer]
I've always had a hard time installing cuda directly, so I switched up the root build image and built the trame image myself.

```Dockerfile
echo "Building ${TRAME_DOCKER_TAG}"
docker build \
    --progress=plain \
    --build-arg BASE_IMAGE=nvidia/cuda:11.2.0-cudnn8-runtime-ubuntu20.04 \
    -t ondiag/trame:common \
    ${TRAME_DIR}/docker \
    -f "${TRAME_DIR}/docker/Dockerfile.common"
docker build \
    --progress=plain \
    --build-arg BASE_IMAGE=ondiag/trame:common \
    -t "${TRAME_DOCKER_TAG}" \
    ${TRAME_DIR}/docker \
    -f "${TRAME_DIR}/docker/Dockerfile.pip"
```

`${TRAME_DIR}` points to a `trame` git submodule. Then I use `${TRAME_DOCKER_TAG}` as the base image. I render with VTK OSMESA. So far runs together (can infer on GPU, can render with VTK). I haven't tried EGL.

**Reply by jourdain** [reply]
Thanks for sharing your command lines and it does not seems to bad doing it yourself... ;-)
